daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/PhaseOne.scala

```diff
@@ -718,17 +718,28 @@ private[lf] final class PhaseOne(
             Return(t.TemplateChoiceDefRef(tmplId, chId)(cid, arg))
           }
         }
-      case UpdateExerciseInterface(ifaceId, chId, cid, arg, guard) =>
+      case UpdateExerciseInterface(ifaceId, chId, cid, arg, maybeGuard) =>
         compileExp(env, cid) { cid =>
           compileExp(env, arg) { arg =>
-            compileExp(env, guard) { guard =>
-              Return(
-                t.InterfaceChoiceDefRef(ifaceId, chId)(
-                  guard,
-                  cid,
-                  arg,
+            maybeGuard match {
+              case Some(guard) =>
+                compileExp(env, guard) { guard =>
+                  Return(
+                    t.InterfaceChoiceDefRef(ifaceId, chId)(
+                      guard,
+                      cid,
+                      arg,
+                    )
+                  )
+                }
+              case None =>
+                Return(
+                  t.InterfaceChoiceDefRef(ifaceId, chId)(
+                    SEAbs(1, SEValue.True),
+                    cid,
+                    arg,
+                  )
```

Comment:
```suggestion
            def choiceDefRef(guard: SExpr) = 
              Return(t.InterfaceChoiceDefRef(ifaceId, chId)(
                      guard,
                      cid,
                      arg,
                    )
                  )
             maybeGuard match {
             case Some(guard) => compileExp(env, guard) choiceDefGuard(_)
             case None => choiceDefguard(SeAbs(1, SEValue.True)
             }
```
maybe something like that is a bit easier to read?

Reply:
:+1: